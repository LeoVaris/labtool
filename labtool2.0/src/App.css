.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.error {
  background-color: red;
  padding: 1em;
  text-align: center;
  color: white;
  font-size: 20;
}

.success {
  background-color: green;
  padding: 1em;
  text-align: center;
  color: white;
  font-size: 20;
}

.TableRowForDroppedOutStudents {
  background-color: lightgray;
}

@media (min-width: 1000px) {
  .mainContainer {
    margin-left: 20px;
    margin-right: 20px;
  }
}
