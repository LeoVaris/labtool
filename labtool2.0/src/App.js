import React, { Component } from 'react'
import './App.css'
import Login from './components/pages/LoginPage'
import Etusivu from './components/pages/MainPage'
import axios from 'axios'

const Notification = ({ message }) => {
  if (message === null) {
    return null
  }
  return (
    <div className="error">
      {message}
    </div>
  )
}

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      loggedIn: false,
      username: '',
      password: '',
      error: ''
    }

    this.changeUserState = this.changeUserState.bind(this)
  }

  changeUserState() {
    this.setState({ loggedIn: !this.state.loggedIn })
  }

  handlePasswordChange = (event) => {
    this.setState({ password: event.target.value })
  }

  handleUsernameChange = (event) => {
    this.setState({ username: event.target.value })
  }

  postLogin = (event) => {

    event.preventDefault()
    
    if (this.state.password !== "" && this.state.username !== "") {
      axios.post('https://opetushallinto.cs.helsinki.fi/login', {
        username: this.state.username,
        password: this.state.password
      })
        .then(response => {
          if (!response.data.error) {
            this.setState({ loggedIn: true })
            console.log('You have succesfully logged in')
            this.setState({ error: '' })

          } else {
            this.setState({ error: 'Wrong username or password' })
            console.log('Wrong username or password')

          }
          this.setState({
            username: '',
            password: ''
          })
        })
        .catch(error => {
          this.setState({
            username: '',
            password: ''
          })
        })
    } else {
      this.setState({error: 'No username or password given'}) 
    }

  }

  render() {
    const u = this.state.username
    const p = this.state.password
    let page = this.state.loggedIn ?
      <Etusivu logout={this.changeUserState} /> :

      <Login
        username={u}
        password={p}
        login={this.changeUserState}
        postLogin={this.postLogin}
        handlePasswordChange={this.handlePasswordChange}
        handleUsernameChange={this.handleUsernameChange}
      />

    return (
      <div className="App" >
        {page}
        <Notification message={this.state.error} />
      </div>
    )
  }
}

export default App
