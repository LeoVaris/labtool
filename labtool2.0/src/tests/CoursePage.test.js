import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'
import { CoursePage } from '../components/pages/CoursePage'
import ConnectedCoursePage from '../components/pages/CoursePage'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Form } from 'semantic-ui-react'
import { getOneCI } from '../services/courseInstance'

describe('<CoursePage />', () => {
  let wrapper
  let output = {
    id: 10011,
    name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
    active: true,
    weekAmount: 7,
    weekMaxPoints: 8,
    currentWeek: 1,
    ohid: 'TKT20010.2018.K.A.1'
  }
  let student = {
    role: 'student',
    data: {
      id: 10011,
      github: 'http://github.com/tiralabra1',
      projectName: 'Tiran labraprojekti',
      courseInstanceId: 10011,
      userId: 10011,
      teacherInstanceId: 10011,
      weeks: [
        {
          id: 10002,
          points: 2,
          weekNumber: 2,
          feedback: 'Melko hienoa työtä!',
          studentInstanceId: 10011,
          comments: []
        },
        {
          id: 10001,
          points: 2,
          weekNumber: 1,
          feedback: 'yvä',
          studentInstanceId: 10011,
          comments: []
        },
        {
          id: 10003,
          points: 3,
          weekNumber: 3,
          feedback: 'Erittäin hienoa työtä!',
          studentInstanceId: 10011,
          comments: []
        },
        {
          id: 10004,
          points: 3,
          weekNumber: 4,
          feedback: 'Hyvin menee!',
          studentInstanceId: 10011,
          comments: []
        }
      ],
      User: {
        id: 10011,
        username: 'tiraopiskelija1',
        email: 'maarit.opiskelija@helsinki.fi',
        firsts: 'Maarit Mirja',
        lastname: 'Opiskelija',
        studentNumber: '014578343',
        admin: false
      }
    }
  }
  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<CoursePage createOneComment={mockFn} getOneCI={mockFn} coursePageInformation={mockFn} associateTeacherToStudent={mockFn} />)
  })

  describe('Course Page Component', () => {
    it('is ok', () => {
      true
    })

    it('should render without throwing an error', () => {
      expect(wrapper.find('.CoursePage').exists()).toEqual(true)
    })
  })
})
