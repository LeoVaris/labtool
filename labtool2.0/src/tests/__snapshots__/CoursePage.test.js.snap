// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoursePage /> as student CoursePage Component should render correctly 1`] = `
<div
  key="true"
>
  <div
    className="StudentsView"
    style={
      Object {
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    <CoursePageHeader
      courseInstance={
        Object {
          "data": Object {
            "Tags": Array [],
            "User": Object {
              "admin": false,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "email": "maarit.opiskelija@helsinki.fi",
              "firsts": "Maarit Mirja",
              "id": 10011,
              "lastname": "Opiskelija",
              "studentNumber": "014578343",
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "username": "tiraopiskelija1",
            },
            "codeReviews": Array [
              Object {
                "points": 2,
                "reviewNumber": 1,
                "reviewer": Object {
                  "github": "http://github.com/superprojekti",
                  "projectName": "Tira super projekti",
                },
                "toReview": Object {
                  "github": "http://github.com/tiralabra2",
                  "projectName": "Tiran toinen labraprojekti",
                },
              },
              Object {
                "points": 1,
                "reviewNumber": 2,
                "reviewer": Object {
                  "github": "http://github.com/tiralabra2",
                  "projectName": "Tiran toinen labraprojekti",
                },
                "toReview": Object {
                  "github": "http://github.com/superprojekti",
                  "projectName": "Tira super projekti",
                },
              },
              Object {
                "points": null,
                "reviewNumber": 3,
                "reviewer": Object {
                  "github": "http://github.com/tiralabra2",
                  "projectName": "Tiran toinen labraprojekti",
                },
                "toReview": Object {
                  "github": "http://github.com/superprojekti",
                  "projectName": "Tira super projekti",
                },
              },
            ],
            "courseInstanceId": 10011,
            "createdAt": "2018-03-26T00:00:00.000Z",
            "github": "http://github.com/tiralabra1",
            "id": 10011,
            "projectName": "Tiran labraprojekti",
            "teacherInstanceId": 10011,
            "updatedAt": "2018-03-26T00:00:00.000Z",
            "userId": 10011,
            "weeks": Array [],
          },
          "role": "student",
        }
      }
    />
    <div
      className="grid"
    >
      <div
        className="sixteen wide column"
      >
        <Message
          compact={true}
        >
          <MessageHeader>
            This course has not been activated.
          </MessageHeader>
        </Message>
      </div>
    </div>
  </div>
  <div
    key="student info"
  >
    <Card
      color="yellow"
      fluid={true}
      key="card"
    >
      <CardContent>
        <h2>
          Maarit Mirja
           
          Opiskelija
        </h2>
        <h3>
           
          Tiran labraprojekti
           
        </h3>
        <h3>
          <a
            href="http://github.com/tiralabra1"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://github.com/tiralabra1
          </a>
           
          <Link
            to="/labtool/courseregistration/undefined"
          >
            <Button
              as="button"
              circular={true}
              floated="right"
              icon={
                Object {
                  "color": "orange",
                  "name": "edit",
                  "size": "large",
                }
              }
              size="large"
            />
          </Link>
        </h3>
      </CardContent>
    </Card>
  </div>
  <Accordion
    fluid={true}
    key="0"
    styled={true}
  >
    <AccordionTitle
      active={false}
      className="codeReview"
      index={0}
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="dropdown"
      />
       Code Review 
      1
       
      , points 2
    </AccordionTitle>
    <AccordionContent
      active={false}
    >
      <div
        className="codeReviewExpanded"
      >
        <div
          className="codeReviewPoints"
        >
          <strong>
            Points: 
          </strong>
           
          2
          <br />
           
          <br />
          <strong>
            Project to review: 
          </strong>
          Tiran toinen labraprojekti
          <br />
          <strong>
            GitHub: 
          </strong>
          <a
            href="http://github.com/tiralabra2"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://github.com/tiralabra2
          </a>
          <br />
           
          <br />
          <div />
        </div>
        <p />
      </div>
    </AccordionContent>
  </Accordion>
  <Accordion
    fluid={true}
    key="1"
    styled={true}
  >
    <AccordionTitle
      active={false}
      className="codeReview"
      index={1}
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="dropdown"
      />
       Code Review 
      2
       
      , points 1
    </AccordionTitle>
    <AccordionContent
      active={false}
    >
      <div
        className="codeReviewExpanded"
      >
        <div
          className="codeReviewPoints"
        >
          <strong>
            Points: 
          </strong>
           
          1
          <br />
           
          <br />
          <strong>
            Project to review: 
          </strong>
          Tira super projekti
          <br />
          <strong>
            GitHub: 
          </strong>
          <a
            href="http://github.com/superprojekti"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://github.com/superprojekti
          </a>
          <br />
           
          <br />
          <div />
        </div>
        <div>
          <div>
            <strong>
              Link your review here:
            </strong>
             
            <br />
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormGroup
                inline={true}
              >
                <Input
                  className="form-control1"
                  icon="github"
                  iconPosition="left"
                  name="reviewLink"
                  placeholder="https://github.com/account/repo/issues/number"
                  required={true}
                  style={
                    Object {
                      "minWidth": "25em",
                    }
                  }
                  type="text"
                />
              </FormGroup>
              <FormGroup>
                <Button
                  as="button"
                  color="blue"
                  compact={true}
                  type="submit"
                >
                  Submit
                </Button>
              </FormGroup>
            </Form>
          </div>
        </div>
      </div>
    </AccordionContent>
  </Accordion>
  <Accordion
    fluid={true}
    key="2"
    styled={true}
  >
    <AccordionTitle
      active={true}
      className="codeReview"
      index={2}
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="dropdown"
      />
       Code Review 
      3
       
    </AccordionTitle>
    <AccordionContent
      active={true}
    >
      <div
        className="codeReviewExpanded"
      >
        <div
          className="codeReviewPoints"
        >
          <strong>
            Points: 
          </strong>
           
          Not graded yet
          <br />
           
          <br />
          <strong>
            Project to review: 
          </strong>
          Tira super projekti
          <br />
          <strong>
            GitHub: 
          </strong>
          <a
            href="http://github.com/superprojekti"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://github.com/superprojekti
          </a>
          <br />
           
          <br />
          <div />
        </div>
        <p />
      </div>
    </AccordionContent>
  </Accordion>
  <Accordion
    fluid={true}
    key="total"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
    styled={true}
  >
    <AccordionTitle
      active={true}
      index="total"
    >
      <Icon
        as="i"
        name="check"
      />
      <strong>
         Total Points: 
      </strong>
      3
    </AccordionTitle>
  </Accordion>
</div>
`;

exports[`<CoursePage /> as teacher CoursePage Component should render correctly 1`] = `
<div
  style={
    Object {
      "overflowX": "auto",
      "overflowY": "hidden",
    }
  }
>
  <div
    className="TeachersTopView"
    style={
      Object {
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    <CoursePageHeader
      courseInstance={
        Object {
          "data": Array [
            Object {
              "Tags": Array [
                Object {
                  "color": "red",
                  "id": 20001,
                  "name": "Javascript",
                },
              ],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "johan.studerande@helsinki.fi",
                "firsts": "Johan Wilhelm",
                "id": 10012,
                "lastname": "Studerande",
                "studentNumber": "014553242",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "tiraopiskelija2",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": false,
              "github": "http://github.com/tiralabra2",
              "id": 10012,
              "projectName": "Tiran toinen labraprojekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "userId": 10012,
              "weeks": Array [],
            },
            Object {
              "Tags": Array [
                Object {
                  "color": "grey",
                  "id": 20008,
                  "name": "DROPPED",
                },
                Object {
                  "color": "yellow",
                  "id": 20002,
                  "name": "HTML",
                },
              ],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "teras.henkilo@helsinki.fi",
                "firsts": "Teräs",
                "id": 10031,
                "lastname": "Henkilö",
                "studentNumber": "014666666",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "superopiskelija",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": true,
              "github": "http://github.com/superprojekti",
              "id": 10031,
              "projectName": "Tira super projekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-06-05T07:12:28.603Z",
              "userId": 10031,
              "weeks": Array [],
            },
            Object {
              "Tags": Array [],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "maarit.opiskelija@helsinki.fi",
                "firsts": "Maarit Mirja",
                "id": 10011,
                "lastname": "Opiskelija",
                "studentNumber": "014578343",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "tiraopiskelija1",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": false,
              "github": "http://github.com/tiralabra1",
              "id": 10011,
              "projectName": "Tiran labraprojekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "userId": 10011,
              "weeks": Array [],
            },
          ],
          "role": "teacher",
        }
      }
    />
    <Table
      as="table"
    >
      <TableHeader
        as="thead"
      >
        <TableRow
          as="tr"
          cellAs="td"
        >
          <TableCell
            as="td"
          >
            <div>
              <div />
            </div>
          </TableCell>
          <TableCell
            as="td"
          >
            Week amount: 
          </TableCell>
          <TableCell
            as="td"
          >
            Current week: 
          </TableCell>
          <TableCell
            as="td"
          >
            Week max points: 
          </TableCell>
          <TableCell
            as="td"
            textAlign="right"
          >
             
            <Link
              to="/labtool/ModifyCourseInstancePage/undefined"
            >
              <Popup
                content="Edit course"
                disabled={false}
                eventsEnabled={true}
                offset={0}
                on={
                  Array [
                    "click",
                    "hover",
                  ]
                }
                pinned={false}
                position="top left"
                trigger={
                  <Button
                    as="button"
                    circular={true}
                    icon={
                      Object {
                        "color": "orange",
                        "name": "edit",
                        "size": "large",
                      }
                    }
                    size="tiny"
                  />
                }
              />
            </Link>
          </TableCell>
        </TableRow>
      </TableHeader>
    </Table>
  </div>
  <div
    className="TeachersBottomViewForActiveStudents"
  >
    <Header
      as="h2"
    >
      Students
       
    </Header>
    <Connect(StudentTable)
      allowModify={true}
      columns={
        Array [
          "select",
          "points",
          "instructor",
          "review",
        ]
      }
      coursePageLogic={
        Object {
          "filterByAssistant": 0,
          "filterByTag": Array [],
          "selectedStudents": Object {
            "10011": true,
          },
          "selectedTeacher": "",
          "showDropdown": "",
        }
      }
      key="TableRowForActiveStudents"
      rowClassName="TableRowForActiveStudents"
      selectedInstance={
        Object {
          "data": Array [
            Object {
              "Tags": Array [
                Object {
                  "color": "red",
                  "id": 20001,
                  "name": "Javascript",
                },
              ],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "johan.studerande@helsinki.fi",
                "firsts": "Johan Wilhelm",
                "id": 10012,
                "lastname": "Studerande",
                "studentNumber": "014553242",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "tiraopiskelija2",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": false,
              "github": "http://github.com/tiralabra2",
              "id": 10012,
              "projectName": "Tiran toinen labraprojekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "userId": 10012,
              "weeks": Array [],
            },
            Object {
              "Tags": Array [
                Object {
                  "color": "grey",
                  "id": 20008,
                  "name": "DROPPED",
                },
                Object {
                  "color": "yellow",
                  "id": 20002,
                  "name": "HTML",
                },
              ],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "teras.henkilo@helsinki.fi",
                "firsts": "Teräs",
                "id": 10031,
                "lastname": "Henkilö",
                "studentNumber": "014666666",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "superopiskelija",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": true,
              "github": "http://github.com/superprojekti",
              "id": 10031,
              "projectName": "Tira super projekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-06-05T07:12:28.603Z",
              "userId": 10031,
              "weeks": Array [],
            },
            Object {
              "Tags": Array [],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "maarit.opiskelija@helsinki.fi",
                "firsts": "Maarit Mirja",
                "id": 10011,
                "lastname": "Opiskelija",
                "studentNumber": "014578343",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "tiraopiskelija1",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": false,
              "github": "http://github.com/tiralabra1",
              "id": 10011,
              "projectName": "Tiran labraprojekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "userId": 10011,
              "weeks": Array [],
            },
          ],
          "role": "teacher",
        }
      }
      studentInstances={
        Array [
          Object {
            "Tags": Array [
              Object {
                "color": "red",
                "id": 20001,
                "name": "Javascript",
              },
            ],
            "User": Object {
              "admin": false,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "email": "johan.studerande@helsinki.fi",
              "firsts": "Johan Wilhelm",
              "id": 10012,
              "lastname": "Studerande",
              "studentNumber": "014553242",
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "username": "tiraopiskelija2",
            },
            "codeReviews": Array [],
            "courseInstanceId": 10011,
            "createdAt": "2018-03-26T00:00:00.000Z",
            "dropped": false,
            "github": "http://github.com/tiralabra2",
            "id": 10012,
            "projectName": "Tiran toinen labraprojekti",
            "teacherInstanceId": 10011,
            "updatedAt": "2018-03-26T00:00:00.000Z",
            "userId": 10012,
            "weeks": Array [],
          },
          Object {
            "Tags": Array [],
            "User": Object {
              "admin": false,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "email": "maarit.opiskelija@helsinki.fi",
              "firsts": "Maarit Mirja",
              "id": 10011,
              "lastname": "Opiskelija",
              "studentNumber": "014578343",
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "username": "tiraopiskelija1",
            },
            "codeReviews": Array [],
            "courseInstanceId": 10011,
            "createdAt": "2018-03-26T00:00:00.000Z",
            "dropped": false,
            "github": "http://github.com/tiralabra1",
            "id": 10011,
            "projectName": "Tiran labraprojekti",
            "teacherInstanceId": 10011,
            "updatedAt": "2018-03-26T00:00:00.000Z",
            "userId": 10011,
            "weeks": Array [],
          },
        ]
      }
      tags={
        Object {
          "tags": Array [
            Object {
              "color": "red",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20001,
              "name": "Javascript",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "yellow",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20002,
              "name": "HTML",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "black",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20003,
              "name": "game",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "green",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20004,
              "name": "React",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "blue",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20005,
              "name": "Node.js",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "orange",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20006,
              "name": "Java",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "pink",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20007,
              "name": "FORTRAN",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "grey",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20008,
              "name": "DROPPED",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
          ],
        }
      }
    />
    <br />
    <Button
      as="button"
      onClick={[Function]}
      size="small"
    >
      Mark all with dropped tag as dropped out
    </Button>
    <Link
      to="/labtool/massemail/undefined"
    >
      <Button
        as="button"
        size="small"
      >
        Send email to multiple students
      </Button>
    </Link>
  </div>
  <br />
  <div
    className="TeachersBottomViewForDroppedOutStudents"
  >
    <Header
      as="h2"
    >
      Dropped out students
       
    </Header>
    <Connect(StudentTable)
      allowModify={true}
      columns={
        Array [
          "select",
          "points",
          "instructor",
          "review",
        ]
      }
      coursePageLogic={
        Object {
          "filterByAssistant": 0,
          "filterByTag": Array [],
          "selectedStudents": Object {
            "10011": true,
          },
          "selectedTeacher": "",
          "showDropdown": "",
        }
      }
      key="TableRowForDroppedOutStudents"
      rowClassName="TableRowForDroppedOutStudents active"
      selectedInstance={
        Object {
          "data": Array [
            Object {
              "Tags": Array [
                Object {
                  "color": "red",
                  "id": 20001,
                  "name": "Javascript",
                },
              ],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "johan.studerande@helsinki.fi",
                "firsts": "Johan Wilhelm",
                "id": 10012,
                "lastname": "Studerande",
                "studentNumber": "014553242",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "tiraopiskelija2",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": false,
              "github": "http://github.com/tiralabra2",
              "id": 10012,
              "projectName": "Tiran toinen labraprojekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "userId": 10012,
              "weeks": Array [],
            },
            Object {
              "Tags": Array [
                Object {
                  "color": "grey",
                  "id": 20008,
                  "name": "DROPPED",
                },
                Object {
                  "color": "yellow",
                  "id": 20002,
                  "name": "HTML",
                },
              ],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "teras.henkilo@helsinki.fi",
                "firsts": "Teräs",
                "id": 10031,
                "lastname": "Henkilö",
                "studentNumber": "014666666",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "superopiskelija",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": true,
              "github": "http://github.com/superprojekti",
              "id": 10031,
              "projectName": "Tira super projekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-06-05T07:12:28.603Z",
              "userId": 10031,
              "weeks": Array [],
            },
            Object {
              "Tags": Array [],
              "User": Object {
                "admin": false,
                "createdAt": "2018-03-26T00:00:00.000Z",
                "email": "maarit.opiskelija@helsinki.fi",
                "firsts": "Maarit Mirja",
                "id": 10011,
                "lastname": "Opiskelija",
                "studentNumber": "014578343",
                "updatedAt": "2018-03-26T00:00:00.000Z",
                "username": "tiraopiskelija1",
              },
              "codeReviews": Array [],
              "courseInstanceId": 10011,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "dropped": false,
              "github": "http://github.com/tiralabra1",
              "id": 10011,
              "projectName": "Tiran labraprojekti",
              "teacherInstanceId": 10011,
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "userId": 10011,
              "weeks": Array [],
            },
          ],
          "role": "teacher",
        }
      }
      studentInstances={
        Array [
          Object {
            "Tags": Array [
              Object {
                "color": "grey",
                "id": 20008,
                "name": "DROPPED",
              },
              Object {
                "color": "yellow",
                "id": 20002,
                "name": "HTML",
              },
            ],
            "User": Object {
              "admin": false,
              "createdAt": "2018-03-26T00:00:00.000Z",
              "email": "teras.henkilo@helsinki.fi",
              "firsts": "Teräs",
              "id": 10031,
              "lastname": "Henkilö",
              "studentNumber": "014666666",
              "updatedAt": "2018-03-26T00:00:00.000Z",
              "username": "superopiskelija",
            },
            "codeReviews": Array [],
            "courseInstanceId": 10011,
            "createdAt": "2018-03-26T00:00:00.000Z",
            "dropped": true,
            "github": "http://github.com/superprojekti",
            "id": 10031,
            "projectName": "Tira super projekti",
            "teacherInstanceId": 10011,
            "updatedAt": "2018-06-05T07:12:28.603Z",
            "userId": 10031,
            "weeks": Array [],
          },
        ]
      }
      tags={
        Object {
          "tags": Array [
            Object {
              "color": "red",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20001,
              "name": "Javascript",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "yellow",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20002,
              "name": "HTML",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "black",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20003,
              "name": "game",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "green",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20004,
              "name": "React",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "blue",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20005,
              "name": "Node.js",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "orange",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20006,
              "name": "Java",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "pink",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20007,
              "name": "FORTRAN",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
            Object {
              "color": "grey",
              "createdAt": "2018-06-13T00:00:00.000Z",
              "id": 20008,
              "name": "DROPPED",
              "updatedAt": "2018-06-13T00:00:00.000Z",
            },
          ],
        }
      }
    />
    <br />
  </div>
  <div
    className="TeacherBulkForm"
  >
    <br />
    <h2>
      Modify selected students
    </h2>
    <Grid
      columns={3}
      divided={true}
      style={
        Object {
          "width": "90%",
        }
      }
    >
      <GridRow>
        <GridColumn>
          <Segment>
            Add/remove tag
          </Segment>
        </GridColumn>
        <GridColumn>
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            closeOnEscape={true}
            deburr={false}
            fluid={true}
            icon="dropdown"
            id="tagDropdown"
            minCharacters={1}
            noResultsMessage="No results found."
            onChange={[Function]}
            openOnFocus={true}
            options={
              Array [
                Object {
                  "key": 20001,
                  "text": "Javascript",
                  "value": 20001,
                },
                Object {
                  "key": 20002,
                  "text": "HTML",
                  "value": 20002,
                },
                Object {
                  "key": 20003,
                  "text": "game",
                  "value": 20003,
                },
                Object {
                  "key": 20004,
                  "text": "React",
                  "value": 20004,
                },
                Object {
                  "key": 20005,
                  "text": "Node.js",
                  "value": 20005,
                },
                Object {
                  "key": 20006,
                  "text": "Java",
                  "value": 20006,
                },
                Object {
                  "key": 20007,
                  "text": "FORTRAN",
                  "value": 20007,
                },
                Object {
                  "key": 20008,
                  "text": "DROPPED",
                  "value": 20008,
                },
              ]
            }
            placeholder="Choose tag"
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            selection={true}
            style={
              Object {
                "float": "left",
              }
            }
            wrapSelection={true}
          />
        </GridColumn>
        <GridColumn>
          <div
            className="two ui buttons"
            style={
              Object {
                "float": "left",
              }
            }
          >
            <button
              className="ui icon positive button"
              onClick={[Function]}
              size="mini"
            >
              <i
                className="plus icon"
              />
            </button>
            <div
              className="or"
            />
            <button
              className="ui icon button"
              onClick={[Function]}
              size="mini"
            >
              <i
                className="trash icon"
              />
            </button>
          </div>
        </GridColumn>
      </GridRow>
      <GridRow>
        <GridColumn>
          <Segment>
            Assign instructor
          </Segment>
        </GridColumn>
        <GridColumn>
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            closeOnEscape={true}
            deburr={false}
            fluid={true}
            icon="dropdown"
            id="assistantDropdown"
            minCharacters={1}
            noResultsMessage="No results found."
            onChange={[Function]}
            openOnFocus={true}
            options={Array []}
            placeholder="Select teacher"
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            selection={true}
            wrapSelection={true}
          />
        </GridColumn>
        <GridColumn>
          <Button
            as="button"
            onClick={[Function]}
            size="small"
          >
            Change instructor
          </Button>
        </GridColumn>
      </GridRow>
      <GridRow>
        <GridColumn>
          <Segment>
            Dropped status
          </Segment>
        </GridColumn>
        <GridColumn>
          <Button
            as="button"
            onClick={[Function]}
          >
            Mark as non-dropped
          </Button>
        </GridColumn>
        <GridColumn>
          <Button
            as="button"
            color="red"
            onClick={[Function]}
          >
            Mark as dropped
          </Button>
        </GridColumn>
      </GridRow>
    </Grid>
    <br />
  </div>
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
</div>
`;
